### PR TITLE
Migrating Serialization, Utils, Crypto, and Identifiers to new repo r…

### PR DESCRIPTION
**Pull request by Sajjon**
…adixdlt-java-common, merged with code from radix-engine-library. Updated code accordingly.

### CODE REVIEW

File: radixdlt-java/build.gradle
Comment (flotothemoon) [inline radixdlt-java/build.gradle]: This doesn't seem right to me, why does this belong to the common project?

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: Would be a good idea to put the exception as a cause rather than just put it in the string. Specifically, I suggest replacing
`new IllegalStateException("Expected to always be able to encrypt, but got exception: " + e)` 
with
 `new IllegalStateException("Expected to always be able to encrypt, but got exception", e)`

File: radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensAction.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensAction.java:26]: Why move RadixAddress and RRI to another package?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java]: Same as above regarding exception as cause

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (flotothemoon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: The same applies to all other new exceptions that were added

File: radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java:74]: Why does `SerializationException` require handling now? It clutters our code in many places.

File: radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWork.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWork.java]: I would prefer `Hash.of` here, but your call

**Review by flotothemoon — CHANGES_REQUESTED** [review]
Good first go. There are some questions regarding the strategy of these PRS as well as issues with the handling of failure cases, formatting and idiomatic use across these PRs that should be addressed before merging.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: great, will change!

File: radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensAction.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensAction.java:26]: The idea being that "Java Common" does not know anything about the AtomModel (yet?). An address is just a formatting of a public key (plus magic byte), it is not dependent or part of the Atom model per se. Neither is the `RRI`, which really just is an identifier, like `EUID`. 

Also package "*.client.*" does not really make sense inside the Java Common. 

So I thought it more correct to use `com.radixdlt.identifiers`

File: radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java:74]: Good question! The serialization implementation in Java lib and Engine differed! @clstrfsck any input? Should I make `fromJson` non throwing? I understand @flotothemoon s point of cluttering the code. However, I'm very much for propagating errors and letting them float up to as high level as possible, rather than crashing "deep down"/low level.

File: radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWork.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWork.java]: agreed! thanks!

File: radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java
Comment (clstrfsck) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java:74]: Hmm.  I think the difference here is the definition of SerializationException.  In radixdlt-java, SerializationException extends RuntimeException, but in radix-engine-library it extends IOException.

I really think the least likely to introduce bugs option here is to have SerializationException extend IOException.  Having it extend RuntimeException has the potential to introduce issues in the engine where catching IOException expects to catch SerializationException.

Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java:74]: @clstrfsck I.e. keeping `SerializationException` as it currently is implemented in this repo... 
So that means that I do need to keep the catch and crash (by throwing `IllegalStateException`) above, or do some other change...? Can the `Suppliers.memoize` be throwing?? 

Sounds like @flotothemoon and @clstrfsck need to agree on a strategy :)

File: radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensAction.java
Comment (clstrfsck) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensAction.java:26]: Also needs to be where serialization is, as address and rri have special serialized forms.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: In my opinion, it's cleaner to use String.format for longer formatted strings since it makes the intended output more obvious and condensed. Same goes for all other exception strings.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: I don't like this way of generating new key pairs. It's non-obvious that this constructor generates a new key-pair rather than using some default, which is an important detail that is lost in this change.

Comment (flotothemoon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: At the least, we should abandon the blank ECKeyPair constructor in favour of a static ECKeyPair.generateKeyPair method.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticleGroupsMapper.java
Comment (flotothemoon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticleGroupsMapper.java]: Curiously, this exception for failing to encrypt something has a different message than the other exception about failing to encrypt something (which says "Expected to always be able to encrypt"). Why are the expectations different?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java
Comment (flotothemoon) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java:74]: I thought we had agreed on throwing a checked (i.e. must-handle-in-some-way) exception for deserialization since the caller must expect this to go wrong. However, below an IllegalStateException is used.

File: radixdlt-java/src/test/java/com/radixdlt/client/core/pow/ProofOfWorkBuilderTest.java
Comment (flotothemoon) [inline radixdlt-java/src/test/java/com/radixdlt/client/core/pow/ProofOfWorkBuilderTest.java]: This does not conform to our styleguide, use `lowerCamelCase` for methods.

File: radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/BootstrapByTrustedNode.java:74]: Yeah going to be changed in coming PR.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: Sure, can fix some, but not really related to this PR.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: Why? What would a "default keypair be"? Why should we treat this class different than any other POJO?

File: radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticleGroupsMapper.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticleGroupsMapper.java]: No reason, this is what happens when using constructor taking a string :P can change to:

```java
throw new IllegalStateException("Expected to always be able to encrypt", e);
```

But harmonizing exception will be done in coming PR.

File: radixdlt-java/src/test/java/com/radixdlt/client/core/pow/ProofOfWorkBuilderTest.java
Comment (Sajjon) [reply radixdlt-java/src/test/java/com/radixdlt/client/core/pow/ProofOfWorkBuilderTest.java]: sure, to be fair we allow snake case for descriptive tests "assert_that_x_do_y" etc. But will change this one!

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (flotothemoon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: Because it's not any other POJO. My point is that a default keypair doesn't make much sense, hence this constructor shouldn't exist (since it's the *default* constructor). Having `new ECKeyPair().equals(new ECKeyPair()) == false` is non-obvious and unexpected, which is bad.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java
Comment (flotothemoon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: True, but since it's already being changed anyway :)

File: radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java
Comment (Sajjon) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java]: Aha ok, I see what you mean, will change to static!

**Comment by Sajjon** [comment]
Fixed gradle with jitpack, so ready for last review and to be merged

**Review by clstrfsck — APPROVED** [review]
Looks good to me.

File: radixdlt-java/src/main/java/com/radixdlt/client/core/address/RadixUniverseConfig.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/address/RadixUniverseConfig.java:148]: Fix me

File: radixdlt-java/src/main/java/com/radixdlt/client/core/address/RadixUniverseConfig.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/address/RadixUniverseConfig.java]: Correct?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/Atom.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/Atom.java]: correct?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/particles/Particle.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/particles/Particle.java]: correct?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/AtomEvent.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/AtomEvent.java:33]: OK?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWork.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWork.java:59]: correct?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWorkBuilder.java
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/pow/ProofOfWorkBuilder.java:55]: correct?

**Comment by Sajjon** [comment]
@clstrfsck inherited low code coverage.... so should we ignore Sonarcloud bot and merge anyway?

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**
